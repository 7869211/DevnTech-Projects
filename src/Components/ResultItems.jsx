import React from 'react';
import '../Styles/ResultItems.css';
import VectorIcon from '../Icons/VectorIcon';

const ResultItems = ({ results }) => {
  const handleTabButton = (event, url) => {
    if (event.key === 'Enter' || event.key === 'click') {
      window.open(url, '_blank');
    }
  };

  const handleClick = (url) => {
    window.open(url, '_blank');
  };

  return (
    <div className="result-items">
      {results.length > 0 ? (
        results.map((item, index) => (
          <div
            key={index}
            className="result-item"
            tabIndex={0}
            onKeyDown={(event) => handleTabButton(event, item.url)}
            onClick={() => handleClick(item.url)}
            role="button"
            aria-label={`Open details for ${item.title}`}
          >
            <img src={item.image} alt={item.title} />
            <div className="result-item-content">
              <div className="result-item-header">
                <h3>{item.title}</h3>
                <VectorIcon />
              </div>
              <p className="description-para">{item.description}</p>
            </div>
          </div>
        ))
      ) : (
        <p className="hidden"></p> 
      )}
    </div>
  );
};

export default ResultItems;
