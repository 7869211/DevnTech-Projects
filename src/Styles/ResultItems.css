.result-items {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  padding: 1rem;
  background-color: white;
  border-radius: 0.2rem;
  overflow-y: auto;
  margin-left: -1rem;
  width: 28.5rem;
}

.result-item {
  display: flex;
  align-items: flex-start;
  padding: 0.8rem;
  background-color: white;
  cursor: pointer;
}

.hidden {
  opacity: 0;
  pointer-events: none;
  display: none;
  visibility: hidden;
}

.result-item:hover {
  background-color: #f2f4f8;
}

.result-item img {
  width: 2.5rem;
  height: 2.5rem;
  object-fit: contain;
  margin-right: 1rem;
}

.result-item-content {
  flex: 1;
  display: flex;
  flex-direction: column;
}

.result-item-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 0.3rem;
}

.result-item h3 {
  font-size: 1.2rem;
  color: #2c3e50;
  margin: 0;
}

.result-item-header h3 {
  flex: 1;
  margin-right: 1rem; 
}

.description-para {
  font-size: 0.875rem;
  color: #7f8c8d;
  margin: 0;
  line-height: 1.4;
}

@media (max-width: 48rem) {
  .result-items {
    max-height: 18.75rem;
  }
}

@media (max-width: 30rem) {
  .result-items {
    padding: 0.5rem;
  }

  .result-item {
    padding: 0.5rem;
    flex-direction: column;
    align-items: flex-start;
  }

  .result-item img {
    width: 2rem;
    height: 2rem;
    margin-right: 0.5rem;
  }

  .result-item-content h3 {
    font-size: 1rem;
  }

  .result-item-header h3 {
    margin-right: 0.5rem;
  }
}
