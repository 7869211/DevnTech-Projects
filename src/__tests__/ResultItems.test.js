import React from 'react';
import { render } from '@testing-library/react';
import ResultItems from '../Components/ResultItems';

jest.mock('../Icons/VectorIcon', () => () => <div data-testid="vector-icon">Mocked Vector Icon</div>);

const mockResults = [
  {
    title: 'Test Title 1',
    description: 'Test Description 1',
    image: 'test-image-1.jpg',
    url: 'https://example.com/1',
  },
  {
    title: 'Test Title 2',
    description: 'Test Description 2',
    image: 'test-image-2.jpg',
    url: 'https://example.com/2',
  },
];

test('ResultItems renders correctly with VectorIcon', () => {
  const { asFragment } = render(<ResultItems results={mockResults} />);
  expect(asFragment()).toMatchSnapshot();
});
