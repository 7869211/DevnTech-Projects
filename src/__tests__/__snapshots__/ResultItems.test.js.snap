// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItems renders correctly 1`] = `
<DocumentFragment>
  <div
    class="result-items"
  >
    <div
      aria-label="Open details for Test Title 1"
      class="result-item"
      role="button"
      tabindex="0"
    >
      <img
        alt="Test Title 1"
        src="test-image-1.jpg"
      />
      <div
        class="result-item-content"
      >
        <h3>
          Test Title 1
        </h3>
        <p
          class="description-para"
        >
          Test Description 1
        </p>
      </div>
    </div>
    <div
      aria-label="Open details for Test Title 2"
      class="result-item"
      role="button"
      tabindex="0"
    >
      <img
        alt="Test Title 2"
        src="test-image-2.jpg"
      />
      <div
        class="result-item-content"
      >
        <h3>
          Test Title 2
        </h3>
        <p
          class="description-para"
        >
          Test Description 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ResultItems renders correctly with VectorIcon 1`] = `
<DocumentFragment>
  <div
    class="result-items"
  >
    <div
      aria-label="Open details for Test Title 1"
      class="result-item"
      role="button"
      tabindex="0"
    >
      <img
        alt="Test Title 1"
        src="test-image-1.jpg"
      />
      <div
        class="result-item-content"
      >
        <div
          class="result-item-header"
        >
          <h3>
            Test Title 1
          </h3>
          <div
            data-testid="vector-icon"
          >
            Mocked Vector Icon
          </div>
        </div>
        <p
          class="description-para"
        >
          Test Description 1
        </p>
      </div>
    </div>
    <div
      aria-label="Open details for Test Title 2"
      class="result-item"
      role="button"
      tabindex="0"
    >
      <img
        alt="Test Title 2"
        src="test-image-2.jpg"
      />
      <div
        class="result-item-content"
      >
        <div
          class="result-item-header"
        >
          <h3>
            Test Title 2
          </h3>
          <div
            data-testid="vector-icon"
          >
            Mocked Vector Icon
          </div>
        </div>
        <p
          class="description-para"
        >
          Test Description 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
